packages/is-shallow-equal/objects.js

```diff
@@ -28,7 +28,12 @@ function isShallowEqualObjects( a, b ) {
 
 	while ( i < aKeys.length ) {
```

Comment:
Unrelated: I'm curious of why while was used here instead of a for (that would be more natural for this loop), was it because "while" proved to be more performant?

Reply:
I've played around a little with loop variations, `while` seems to be the most performant (in Node, at least).

Reply:
Probably also worth pointing out that these are very much engine-specific optimizations that I tend to be generally okay with, but also noting that:

- The performance characteristics can (and likely will) change over time
- Ideally they are still equally as performant (or as close to as possible) in other engines
   - In the context of Node at least, optimizing specifically for V8 is reasonable, but in a browser environment, we should be considerate of [SpiderMonkey](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey) and friends